### PR TITLE
Update mod to release-20230225

### LINKED ISSUE #212: Update to use latest OpenRA engine

### PR DESCRIPTION
**Pull request by penev92**
It is finally time to update the mod to OpenRA's release `20230225`.

To make this as simple as possible for everyone involved I tried to follow the commit structure and all the ideas behind https://github.com/OpenRA/ra2/pull/806.
>To ease the review as much as possible this uses the latest ModSDK version from https://github.com/OpenRA/OpenRAModSDK/pull/183, the same as TDHD's release-20230225 uses.

This is still WIP, but I'm opening it so hopefully the reviewing process can gradually get going and also so someone else doesn't waste effort doing the same work at the same time.

Supersedes #201.
Closes #212.

### CODE REVIEW

**Comment by penev92** [comment]
Added a bunch of fixes, updates and polish. This should now be good to go.

**Comment by penev92** [comment]
Ok, apparently I was wrong - this was not ready. Both `make check` and `make test` were going nuts and I hadn't noticed.
So I added a bunch of code and yaml fixes, some fixing issues from many years ago.
It should be good to go *now*.

**Comment by evgeniysergeev** [comment]
> @evgeniysergeev are you able to review? I'm most interested in some things from commit `Manual mod updates PART 9`, like the Devastator overloading.

Sorry for late answer. I'm trying to review now.

File: mods/d2/sequences/misc.yaml
Comment (evgeniysergeev) [inline mods/d2/sequences/misc.yaml:1]: BuildableTerrainOverlay is replaced with https://github.com/OpenRA/d2/blob/master/OpenRA.Mods.D2/Graphics/D2BuildingPlacementRenderable.cs
Screenshot 2023-07-04 225032
Is it still used somewhere?

File: mods/d2/rules/world.yaml
Comment (evgeniysergeev) [inline mods/d2/rules/world.yaml:1]: After the update, spice ends very quickly on maps. In the original game it ends not so fast.

**Comment by evgeniysergeev** [comment]
@penev92 , I think this is great PR. All this comments above can be fixed later, and changes can be merged. But if you have a time, can you take a look at comments?

File: mods/d2/rules/barracks.yaml
Comment (evgeniysergeev) [inline mods/d2/rules/barracks.yaml:1]: Pips were not used in the original game. Is it important to add them in the update?
Issue https://github.com/OpenRA/d2/issues/148 was created to remove pips.

File: mods/d2/mod.yaml
Comment (evgeniysergeev) [inline mods/d2/mod.yaml:1]: Maybe also replace
`	~^SupportDir|maps/d2/release-20210321: User`
with
`	~^SupportDir|maps/d2/release-20230225: User` ?

**Comment by evgeniysergeev** [comment]
> @evgeniysergeev are you able to review? I'm most interested in some things from commit `Manual mod updates PART 9`, like the Devastator overloading.

Don't know what should happen when Devastator is overloaded. Is this overloading exists in original game? :-)

**Review by evgeniysergeev — APPROVED** [review]
Will fix issues from comments after the merge

**Pull request merged**